keps/sig-instrumentation/1661-event-api-migration/README.md

```diff
@@ -0,0 +1,467 @@
+type EventSeries struct {
+  // Number of occurrences in this series up to the last heartbeat time
+  Count int32 `json:"count" protobuf:"varint,1,opt,name=count"`
+  // Time when last Event from the series was seen before last heartbeat.
+  LastObservedTime metav1.MicroTime `json:"lastObservedTime" protobuf:"bytes,2,opt,name=lastObservedTime"`
+  // Information whether this series is ongoing or finished.
+  // Deprecated. Planned removal for 1.18
+  State EventSeriesState `json:"state" protobuf:"bytes,3,opt,name=state"`
+}
+
+type EventSeriesState string
+
+const (
+  EventSeriesStateOngoing  EventSeriesState = "Ongoing"
+  EventSeriesStateFinished EventSeriesState = "Finished"
+  EventSeriesStateUnknown  EventSeriesState = "Unknown"
+)
+```
+
+#### Few Examples
+
+| Regarding | Action | Reason | ReportingController | Related | 
+| ----------| -------| -------| --------------------|---------| 
+| Node X | BecameUnreachable | HeartbeatTooOld | kubernetes.io/node-ctrl | <nil> |
+| Node Y | FailedToAttachVolume | Unknown | kubernetes.io/pv-attach-ctrl | PVC X |
+| ReplicaSet X | FailedToInstantiatePod | QuotaExceeded | kubernetes.io/replica-set-ctrl | <nil> |
+| ReplicaSet X | InstantiatedPod | | kubernetes.io/replica-set-ctrl | Pod Y |
+| Ingress X | CreatedLoadBalancer | | kubernetes.io/ingress-ctrl | <nil> |
+| Pod X | ScheduledOn | | kubernetes.io/scheduler | Node Y |
+| Pod X | FailedToSchedule | FitResourcesPredicateFailed | kubernetes.io/scheduler | <nil> |
+
+#### Comparison Between Old and New APIs
+
+| Old         | New         |
+|-------------|-------------|
+| Old Event { | New Event { |
+| TypeMeta | TypeMeta |
+| ObjectMeta | ObjectMeta |
+| InvolvedObject ObjectReference | Regarding ObjectReference |
+|  | Related *ObjectReference |
+|  | Action string |
+| Reason string | Reason string |
+| Message string | Note string |
+| Source EventSource |  |
+|  | ReportingController string |
+|  | ReportingInstance string |
+| FirstTimestamp metav1.Time | |
+| LastTimestamp metav1.Time | |
+|  | EventTime metav1.MicroTime |
+| Count int32 | |
+|  | Series EventSeries |
+| Type string | Type string |
+| } | } |
+
+Namespace in which Event will live will be equal to
+- Namespace of Regarding object, if it's namespaced,
+- NamespaceSystem, if it's not.
+
+Note that this means that if Event has both Regarding and Related objects, and only one of them is namespaced, it should be used as Regarding object.
+
+The biggest change is the semantics of the Event object in case of loops. If Series is nil it means that Event is a singleton, i.e. it happened only once and the semantics is exactly the same as currently in Events with `count = 1`. If Series is not nil it means that the Event is either beginning or the end of an Event series - equivalence of current Events with `count > 1`. Events for ongoing series have Series.State set to EventSeriesStateOngoing, while endings have Series.State set to EventSeriesStateFinished.
```

Comment:
the state field is flagged as planned for removal in 1.18... is this sentence still accurate?

Reply:
@liggitt - Agreed, we never used it in the first place.

Reply:
Yes - should be removed.

Intead of removing this sentence, I would add an explicit section about fields that were deprecated and should be removed before graduating to GA.
It's worth mentioning that this field should also be removed from corev1.Event too:
https://github.com/kubernetes/kubernetes/pull/75987

Reply:
Added a "Deprecated Fields" section under "Graduation Criteria". Also added a notes for the last sentence.